src/creds/VerifiableCredential.js

```diff
@@ -410,6 +423,61 @@ function VerifiableCredentialBaseConstructor(identifier, issuer, expiryIn, ucas,
     return result;
   };
 
+  /**
+   * Updates the credential with a "granted" token based on the requestorId and a unique requestId (a nonce) that
+   * can be verified later using .verify() function.
+   *
+   * @param  {string} requestorId - The IDR id (DID).
+   * @param  {string} requestId - A unique requestID. This should be a nonce for proof chanlange.
+   * @param  {string} pvtKey - Depending of the provided CryptoManager service, it can be either a
+   *                           keyName (i.e. the CredentialWallet Impl) or a pvtKey in base58 format (the default).
+   */
+  this.grantUsageFor = (requestorId, requestId, pvtKey) => {
+    if (_.isEmpty(_.get(this.proof, 'anchor.subject.label')) || _.isEmpty(_.get(this.proof, 'anchor.subject.pub'))) {
+      throw new Error('Invalid credential attestation/anchor');
+    }
+    if (!this.verifySignature()) {
+      throw new Error('Invalid credential attestation/anchor signature');
+    }
+    if (!requestorId || !requestId || !pvtKey) {
+      throw new Error('Missing required parameter: requestorId, requestId or key');
+    }
+    const stringToHash = `${this.proof.anchor.subject.label}${this.proof.anchor.subject.pub}${requestorId}${requestId}`;
+    const hexHash = sha256(stringToHash);
+
+    const cryptoManager = services.container.CryptoManager;
+
+    const hexSign = cryptoManager.sign(pvtKey, hexHash);
+    this.granted = hexSign;
+  };
+
+  /**
+   * @param  {} requestorId
+   * @param  {} requestId
+   * @param  {} [keyName]
+   */
+  this.verifyGrant = (requestorId, requestId, keyName) => {
+    let verified = false;
+    if (_.isEmpty(_.get(this.proof, 'anchor.subject.label')) || _.isEmpty(_.get(this.proof, 'anchor.subject.pub'))) {
+      return verified;
+    }
+    if (_.isEmpty(this.granted)) {
+      return verified;
+    }
+    if (!requestorId || !requestId) {
+      return verified;
+    }
+    const stringToHash = `${this.proof.anchor.subject.label}${this.proof.anchor.subject.pub}${requestorId}${requestId}`;
+    const hexHash = sha256(stringToHash);
+
+    const cryptoManager = services.container.CryptoManager;
```

Comment:
We had problems with this inside credential wallet please adapt as the injection of secure random on VC class

Reply:
This is correct, I confused with the wrong one